### PR TITLE
[W06] DualPivotQuicksortTests

### PR DESCRIPTION
**Pull request by YifanChen03**
#### General

- [x] I confirm that my tests, if any, are correct
    - [x] Optional: I pass **all** artemis tests.
    - [x] Optional: I pass **all** tests, that are added in this PR.
    - [x] Optional: I calculated manually the correct solutions on paper.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://github.com/JohannesStoehr/gad23-tests/blob/main/CONTRIBUTING.md).
- [x] The newly added tests do not reveal the solutions dynamically calculating the solutions as expected in the exercise.
- [ ] Other students approved this Pull Request.

### CODE REVIEW

**Review by Flnn0 — CHANGES_REQUESTED** [review]
Code LGTM but the part that you have to remove the logging from your original code isn't good. Maybe you could implemten it as myself in PR #35 with the an `StudentResultTest` class. It's much easier for the others and there is no risk to forget to put the logging back in.

**Comment by YifanChen03** [comment]
> Code LGTM but the part that you have to remove the logging from your original code isn't good. Maybe you could implemten it as myself in PR #35 with the an `StudentResultTest` class. It's much easier for the others and there is no risk to forget to put the logging back in.

added a class StudentResultForTest.java to simplify testing

File: w06/test/gad/DualPivotQuicksortTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        for (int i = 0; i < expected.length; i++) {
+            if (i >= from && i <= to) {
+                expectedCopy3[i] = expectedSortedPart[i - from];
+            } else {
+                expectedCopy3[i] = expected[i];
+            }
+        }
+
+        DualPivotQuicksort dualPivotQuicksort = new DualPivotQuicksort(DualPivotFinder.getRandomPivot(10), 0);
+        StudentResult result = new StudentResultForTest();
```
Comment (Flnn0) [inline w06/test/gad/DualPivotQuicksortTest.java:157]: You have to change `StudentResult` to `StudentResultForTest` otherways you get a error.

File: w06/test/gad/DualPivotQuicksortTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    public void dualPivotQuickSortTestLarge() {
+        Random random = new Random();
+        int[] numbers = new int[1_000_000];
+        for (int i = 0; i < numbers.length; i++) {
+            numbers[i] = random.nextInt(Integer.MAX_VALUE);
+        }
+        int[] expected = Arrays.copyOf(numbers, numbers.length);
+
+        DualPivotQuicksort dualPivotQuicksort = new DualPivotQuicksort(DualPivotFinder.getRandomPivot(10), 0);
+        StudentResult result = new StudentResult();
```
Comment (Flnn0) [inline w06/test/gad/DualPivotQuicksortTest.java:107]: Here it's the same as in Line `157`. Especially in this Test the logging is rising the runtime and this caused that some devices might not finish this test.

File: w06/test/gad/StudentResultForTest.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import gad.simplesort.Result;
+
+public class StudentResultForTest implements Result {
+    @Override
+    public void logArray(int[] array) {}
```
Comment (Flnn0) [inline w06/test/gad/StudentResultForTest.java:5]: This isn't the logging method of this exercise. Have a look in `StudentResult` which is given from Artemis.

**Review by Flnn0 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**